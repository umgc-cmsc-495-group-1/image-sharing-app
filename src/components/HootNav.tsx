import React, { useState } from "react";
import {
  AppBar,
  Box,
  Drawer,
  IconButton,
  List,
  ListItem,
  ListItemText,
  Toolbar,
  Typography,
} from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import { Link, Outlet, useNavigate } from "react-router-dom";
import { auth } from "../firebaseSetup";
import { logout } from "../data/authFunctions";
import { onAuthStateChanged } from "firebase/auth";


export default function HootNav() {
  const [uid, setUid] = useState("");

  const navigate = useNavigate();

  onAuthStateChanged(auth, (user) => {
    if (user) {
      setUid(user.uid);
    } else {
      setUid("");
    }
  });

  const [isOpen, setIsOpen] = useState(false);

  function toggleDrawer(
    open: boolean,
    event: React.KeyboardEvent | React.MouseEvent
  ) {
    if (
      event.type === "keydown" &&
      ((event as React.KeyboardEvent).key == "Tab" ||
        (event as React.KeyboardEvent).key == "Shift")
    ) {
      return;
    }

    setIsOpen(open);
  }

  return (
    <div className="class.navigation">
      <Box sx={{ flexGrow: 1 }}>
        <AppBar position="static">
          <Toolbar>
            <IconButton
              size="large"
              edge="start"
              color="inherit"
              aria-label="menu"
              sx={{ mr: 2 }}
              role='menu-icon'
              onClick={(event) => {
                toggleDrawer(!isOpen, event);
              }}
            >
              <MenuIcon />
            </IconButton>
            <Box
              component="img"
              sx={{
                height: 64,
              }}
              alt="Hoot Logo"
              src={require("../assets/logo/png/simple-72x72.png")}
            />
            <Typography variant="h6">
              {uid ? uid + " - Hoot!" : "Hoot!"}
            </Typography>
          </Toolbar>
        </AppBar>
      </Box>
      <Drawer
        anchor="left"
        open={isOpen}
        onClose={() => {
          setIsOpen(false);
        }}
      >
        <Toolbar></Toolbar>
        <Box
          sx={{ width: 250 }}
          role="presentation"
          onClick={(event) => {
            toggleDrawer(false, event);
          }}
          onKeyDown={(event) => {
            toggleDrawer(false, event);
          }}
        >
          <List>
            <ListItem button component={Link} to="/" role="navigation-home">
              <ListItemText primary="Home" />
            </ListItem>
            {!uid ? ( // Logged out User Nav Section
              <>
                <ListItem button component={Link} to="/login" role="navigation-login">
                  <ListItemText primary="Login" />
                </ListItem>
                <ListItem button component={Link} to="/signup" role="navigation-signup">
                  <ListItemText primary="Sign Up" />
                </ListItem>
              </>
            ) : (
              ""
            )}
            {uid ? ( // Logged in User Nav Section
              <>
                <ListItem
                  button
                  onClick={() => {
                    logout();
                    navigate("../", { replace: true });
                  }}
                >
                  <ListItemText primary="Sign Out" />
                </ListItem>
              </>
            ) : (
              ""
            )}
          </List>
        </Box>
      </Drawer>
      <Outlet />
    </div>
  );
}