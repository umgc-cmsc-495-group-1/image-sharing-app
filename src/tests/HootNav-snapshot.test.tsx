import React from 'react';
import "@testing-library/react/dont-cleanup-after-each";
import '@testing-library/jest-dom';
import { 
  MemoryRouter,
  Routes,
  Route
} from 'react-router-dom';
import HootNav from '../components/HootNav';
import { create } from 'react-test-renderer';

function renderWithMemoryRouter(component: JSX.Element) {
  return create(
    <MemoryRouter initialEntries={['/']}>
      <Routes>
        <Route path='/' element={component} />
      </Routes>
    </MemoryRouter>
  );
}

it('Renders correctly',() => {
  const renderer = renderWithMemoryRouter(<HootNav/>);
  expect(renderer.toJSON()).toMatchSnapshot();
})

