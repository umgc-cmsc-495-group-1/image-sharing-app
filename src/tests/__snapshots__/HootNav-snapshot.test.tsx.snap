// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="class.navigation"
>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="menu-icon"
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </button>
        <img
          alt="Hoot Logo"
          className="MuiBox-root css-167lj4j"
          src="simple-72x72.png"
        />
        <h6
          className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Hoot!
        </h6>
      </div>
    </header>
  </div>
</div>
`;
